modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/services/HealthCheckConsensus.scala

```diff
@@ -0,0 +1,41 @@
+package org.tessellation.sdk.domain.healthcheck.services
+
+import cats.Monad
+import cats.effect.Ref
+import cats.syntax.bifunctor._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.traverse._
+
+import org.tessellation.sdk.domain.healthcheck.types.ConsensusRounds
+
+abstract class HealthCheckConsensus[F[_]]()(implicit F: Monad[F]) extends HealthCheck[F] {
+  def allRounds: Ref[F, ConsensusRounds[F]]
+  def roundsInProgress = allRounds.get.map(_.inProgress)
+
+  final override def trigger(): F[Unit] =
+    triggerRound()
+
+  def triggerRound(): F[Unit] =
+    roundsInProgress.flatMap {
+      case inProgress if inProgress.nonEmpty => manageRounds(inProgress)
```

Comment:
Is this pattern match actually needed? I prefer using `if-else` for the simplest cases.
```
roundsInProgress.flatMap { inProgress =>
if (inProgress.nonEmpty)
  manageRounds(inProgress)
else
  Applicative[F].unit
}
```

Reply:
it's just a matter of clarity but else case or more cases will be added later